src/omv/imlib/framebuffer.c

```diff
@@ -227,19 +236,52 @@ uint32_t framebuffer_get_frame_size()
     return image_size(&img);
 }
 
+// Returns the number of bytes the frame buffer could be at the current moment it time.
+static int framebuffer_raw_buffer_size()
+{
+    int size = (int) (fb_alloc_stack_pointer() - ((char *) framebuffer->pixels));
+    // We don't want to give all of the frame buffer RAM to the frame buffer. So, we will limit
+    // the maximum amount of RAM we return.
+    return IM_MIN(size, OMV_RAW_BUF_SIZE);
+}
+
 uint32_t framebuffer_get_buffer_size()
 {
-    uint32_t size = (uint32_t) (fb_alloc_stack_pointer() - ((char *) framebuffer->pixels));
-    // We don't want to give all of the frame buffer RAM to the frame buffer. So, we will limit the
-    // maximum amount of RAM we return.
-    size = IM_MIN(size, OMV_RAW_BUF_SIZE);
+    int size;
+
+    if (vbuffer_count == 1) {
+        // With only 1 vbuffer it's fine to allow the frame buffer size to change given fb_alloc().
+        size = framebuffer_raw_buffer_size();
+    } else {
+        // Whatever the raw size was when the number of buffers were set is locked in...
+        size = vbuffer_raw_size;
+    }
+
+    // Remove the size of the state header plus alignment padding.
+    size -= VBUFFER_STATE_SIZE;
```

Comment:
```suggestion
        // Remove the size of the vbuffer header.
        size = vbuffer_raw_size - sizeof(vbuffer_t);
    }
```
While remove the header for 1 buffer ?

Reply:
The header exists in the 1 buffer case too.